public/app/core/time_series2.ts

```diff
@@ -75,7 +75,12 @@ export function getDataMinMax(data: TimeSeries[]) {
 export default class TimeSeries {
   datapoints: any;
   id: string;
+  // Represents index of original data frame in the quey response
   dataFrameIndex: number;
+  /**
+   * Name of the field the time series was created from
+   * Used in graph panel to retrieve value from the original data frame
+   */
   fieldName: string;
```

Comment:
As I think more... should we really store the fieldIndex?  name does not need to be unique (though it most likely is) and requires scanning all fields to find it

Reply:
as we discussed -> to be replaced with fieldIndex